### PR TITLE
Add Generic Water Heater

### PR DESCRIPTION
**Pull request by dgomes**
## Proposed change

Add documentation of a new Generic Water Heater integration

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/44605
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/generic_water_heater.markdown
Comment (frenck) [inline source/_integrations/generic_water_heater.markdown]: This violates ADR-0007

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/generic_water_heater.markdown
Comment (dgomes) [reply source/_integrations/generic_water_heater.markdown]: https://github.com/home-assistant/example-custom-config/tree/master/custom_components/example_light/ is not longer up to date, right ?

Comment (dgomes) [reply source/_integrations/generic_water_heater.markdown]: refactored

Whats the stance on these generic platforms? should or shouldn't have config flows ?

File: source/_integrations/generic_water_heater.markdown
Comment (vorostamas) [inline source/_integrations/generic_water_heater.markdown]:
```suggestion
ha_release: 2021.3
```

Comment (dgomes) [reply source/_integrations/generic_water_heater.markdown]: Thank you for your review! 

Since PR have been running very slowly I'll wait for core PR approval before correcting this ;)

**Comment by baylanger** [comment]
Thank you for adding this, I've actually ordered all hardware to control my water heater.

[q] Is there a reason to limit this generic water heater to a single temperature sensor? Large Water Heaters have 2 heating elements, upper and lower, each controlled with its own thermostat. The cold water flows in at the bottom of the tank, thus keeping hot water at the top.

It's not always necessary to heat up the whole tank. It greatly depends on how many people are home (presence) and the water intake Temperature (winter vs summer). For those reasons, since we are 4 in here, I'm aiming to install 2 T probes.

I'm not requesting to make any changes to this PR but it would be great in the future to have more logic for such a case.

There's full explanation how dual heater water tank works in this video:
https://www.youtube.com/watch?v=Bm7L-2J52GU

[edit] I just realized this is only for the documentation :)

**Comment by dgomes** [comment]
Mine has only one sensor :(

But please let's not spam the PR

**Comment by frenck** [comment]
Closing PR as the parent PR has been closed.

**Pull request closed without merging**
